### PR TITLE
1.2.0

### PR DESCRIPTION
**Pull request by Lurivar**
- Update query columns with the new COL_ variants, since the others are deprecated
- Added a config page to the module, allowing to choose which order statuses are used to calculate the best sellers. Default is 2,3,4 (paid, processed, sent).

### CODE REVIEW

File: EventListeners/EventManager.php [hunk 1]
```diff
@@ -58,38 +58,38 @@ public function calculateBestSellers(BestSellersEvent $event)
                 /** @var PdoConnection $con */
                 $con = Propel::getConnection();
 
-                $query = "
+                $statusList = BestSellers::getConfigValue('order_types');
```
Comment (roadster31) [inline EventListeners/EventManager.php:61]: Please provide a default value here.

File: Controller/ConfigController.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class ConfigController extends BaseAdminController
+{
+    public function setAction()
+    {
+        $form = new Configuration($this->getRequest());
+        $response = null;
+
+        $configForm = $this->validateForm($form);
+        BestSellers::setConfigValue('order_types', $configForm->get('order')->getData(),null, true);
+
```
Comment (roadster31) [inline Controller/ConfigController.php:33]: It seems a little bit dangerous to store the data entered by the admin without filtering or checking.

File: EventListeners/EventManager.php (hunk 1 above)
Comment (Lurivar) [reply EventListeners/EventManager.php:61]: I've added a check which sets a default value if it fails.

File: Controller/ConfigController.php (hunk 2 above)
Comment (Lurivar) [reply Controller/ConfigController.php:33]: This is now encapsulated in a try/catch.

**Comment by roadster31** [comment]
Thanks 👍

**Pull request merged**